opencensus/exporters/trace/zipkin/zipkin_exporter.cc

```diff
@@ -0,0 +1,342 @@
+
+  writer->Key("timestamp");
+  writer->Int64(absl::ToUnixMicros(span.start_time()));
+
+  writer->Key("duration");
+  writer->Int64(absl::ToInt64Microseconds(span.end_time() - span.start_time()));
+
+  writer->EndObject();
+}
+
+class CurlEnv {
+ public:
+  CurlEnv() { curl_global_init(CURL_GLOBAL_DEFAULT); }
+  ~CurlEnv() { curl_global_cleanup(); }
+};
+
+}  // namespace
+
+std::string JsonCodec::Encode(
+    const std::vector<::opencensus::trace::exporter::SpanData> &spans) {
+  rapidjson::StringBuffer buffer;
+  rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+
+  writer.StartArray();
+  for (auto &span : spans) {
+    SerializeJson(&writer, span);
+  }
+  writer.EndArray();
+  return buffer.GetString();
+}
+
+std::string PrettyJsonCodec::Encode(
+    const std::vector<::opencensus::trace::exporter::SpanData> &spans) {
+  rapidjson::StringBuffer buffer;
+  rapidjson::PrettyWriter<rapidjson::StringBuffer> writer(buffer);
+
+  writer.StartArray();
+  for (auto &span : spans) {
+    SerializeJson(&writer, span);
+  }
+  writer.EndArray();
+  return buffer.GetString();
+}
+
+struct HttpClientOptions {
+  HttpClientOptions(absl::string_view u) : url(u) {}
+
+  // The URL to use in the request.
+  std::string url;
+  // The proxy to use for the upcoming request.
+  std::string proxy;
+  // Tunnel through HTTP proxy
+  bool http_proxy_tunnel = false;
+  // The maximum number of redirects allowed. The default maximum redirect times
+  // is 3.
+  size_t max_redirect_times = 3;
+  // The maximum timeout for TCP connect. The default connect timeout is 5
+  // seconds.
+  absl::Duration connect_timeout = absl::Seconds(5);
+  // The maximum timeout for HTTP request. The default request timeout is 15
+  // seconds.
+  absl::Duration request_timeout = absl::Seconds(15);
+};
+
+class HttpClient : public ExportClient {
+ public:
+  HttpClient(const HttpClientOptions &options)
+      : url_(options.url),
+        proxy_(options.proxy),
+        http_proxy_tunnel_(options.http_proxy_tunnel),
+        max_redirect_times_(options.max_redirect_times),
+        connect_timeout_(options.connect_timeout),
+        request_timeout_(options.request_timeout) {}
+
+  void SendMessage(const std::string &msg, size_t size) override;
+
+ private:
+  std::string url_;
+  std::string proxy_;
+  bool http_proxy_tunnel_;
+  size_t max_redirect_times_;
+  absl::Duration connect_timeout_;
+  absl::Duration request_timeout_;
+};
+
+void HttpClient::SendMessage(const std::string &msg, size_t size) {
+  CURLcode res;
+  struct curl_slist *headers = nullptr;
+  char content_type[128] = {0}, err_msg[CURL_ERROR_SIZE] = {0};
+  std::string error;
+  CURL *curl = curl_easy_init();
+
+  if (!curl) {
+    // Failed to create curl handle.
+    return;
+  }
+
+  const std::string mime_type = "application/json";
+  snprintf(content_type, sizeof(content_type), "Content-Type: %s",
+           mime_type.c_str());
+
+  headers = curl_slist_append(headers, content_type);
```

Comment:
Inline it all.

`headers = curl_slist_append(headers, "Content-Type: application/json");`

Reply:
Done.